src/cli.rs

```diff
@@ -0,0 +1,821 @@
+
+    if let Err(error) = result {
+      println!("{}", error);
+    }
+  }
+}
+
+async fn list(client: impl QuerySecrets, _args: &List) -> Result<String, NysmError> {
+  let secrets_list = client.secrets_list().await?;
+
+  Ok(secrets_list.table_display())
+}
+
+async fn show(client: impl QuerySecrets, args: &Show) -> Result<(), NysmError> {
+  let secret_value = client.secret_value(args.secret_id.clone()).await?;
+  let secret = secret_value.secret;
+
+  let formatted_secret = reformat_data(&secret, &args.secret_format, &args.print_format)?;
+  let _ = pretty_print(formatted_secret, &args.print_format);
+
+  Ok(())
+}
+
+async fn edit(client: impl QuerySecrets, args: &Edit) -> Result<(), NysmError> {
+  let secret_value = client.secret_value(args.secret_id.clone()).await?;
+  let secret = secret_value.secret;
+
+  if let Ok(dir) = temporary_directory() {
+    let update_contents = launch_editor(secret, dir, &args.secret_format, &args.edit_format)?;
+
+    if let Some(contents) = update_contents {
+      let _ = client
+        .update_secret_value(args.secret_id.clone(), contents)
+        .await?;
+    }
+  }
+
+  Ok(())
+}
+
+fn reformat_data(
+  content: &str,
+  source_format: &DataFormat,
+  destination_format: &DataFormat,
+) -> Result<String, NysmError> {
+  Ok(match source_format {
+    DataFormat::Json => {
+      let json_value: serde_json::Value = serde_json::from_str(content)?;
+
+      match destination_format {
+        DataFormat::Json => serde_json::to_string_pretty(&json_value)?,
+        DataFormat::Yaml => serde_yaml::to_string(&json_value)?,
+        DataFormat::Text => String::from(content),
+      }
+    }
+    DataFormat::Yaml => {
+      let yaml_value: serde_yaml::Value = serde_yaml::from_str(content)?;
+
+      match destination_format {
+        DataFormat::Json => serde_json::to_string_pretty(&yaml_value)?,
+        DataFormat::Yaml => serde_yaml::to_string(&yaml_value)?,
+        DataFormat::Text => String::from(content),
+      }
+    }
+    DataFormat::Text => String::from(content),
+  })
+}
+
+#[cfg(not(tarpaulin_include))]
+fn pretty_print(content: String, print_format: &DataFormat) -> Result<(), NysmError> {
+  if std::io::stdout().is_terminal() {
+    let language_string = print_format.to_string();
+    let mut printer = PrettyPrinter::new();
+    let _printer = match print_format {
+      DataFormat::Yaml | DataFormat::Json => printer.language(&language_string),
+      _ => &mut printer,
+    };
+
+    #[allow(unused)]
+    #[cfg(not(test))]
+    let _ = _printer
+      .grid(true)
+      .line_numbers(true)
+      .paging_mode(bat::PagingMode::QuitIfOneScreen)
+      .pager("less")
+      .theme("OneHalfDark")
+      .input_from_bytes(content.as_bytes())
+      .print()?;
+  } else {
+    println!("{}", content);
+  }
+
+  Ok(())
+}
+
+fn launch_editor(
+  contents: String,
+  path: tempdir::TempDir,
+  secret_format: &DataFormat,
+  edit_format: &DataFormat,
+) -> Result<Option<String>, NysmError> {
```

Comment:
Even though this isn't public, some docs about the returned item would be nice.

Reply:
It probably is worth documenting this method. The return is either a string containing the edited contents returned from the edited file or none as the file hasn't changed. Also there is the case where an error is encountered and we return a NysmError to explain what went wrong (likely file io issues).